fili-core/src/main/java/com/yahoo/bard/webservice/table/BaseCompositePhysicalTable.java

```diff
@@ -19,7 +19,7 @@
 /**
  * An implementation of BasePhysicalTable that contains multiple tables.
  */
-public abstract class BaseCompositePhysicalTable extends BasePhysicalTable {
+public class BaseCompositePhysicalTable extends BasePhysicalTable {
```

Comment:
If we make this concrete, then we should deprecate `MetricUnionCompositeTable` and `PartitionCompositeTable` as well because all they were doing were constructing `MetricUnionAvailability` and `PartitionAvailability`, respectively.

The following is not a blocker to this PR:

1. `MetricUnionCompositeTableDefinition` doesn't construct `MetricUnionCompositeTable` anymore since it's been deprecated. Instead, the definition should construct `BaseCompositePhysicalTable`
2. Need a builder method for `MetricUnionAvailability` that builds `MetricUnionAvailability` directly
3. Same tasks(1 & 2) for `DimensionListPartitionTableDefinition`(`PartitionCompositeTable`)